### PR TITLE
ISSUE-3618: Improved performance of arithmetic functions

### LINKED ISSUE #3620: support div intdiv mul minus module, remove the unused code

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- Improvement
- Performance Improvement

## Related Issues

Fixes #3620 

## Test Plan

Unit Tests

### CODE REVIEW

File: common/functions/src/scalars/arithmetics/arithmetic_modulo.rs [hunk 1]
```diff
@@ -45,44 +45,35 @@ pub struct ArithmeticModule<T, D, M, R> {
...
-        let lhs: &DFPrimitiveArray<T> = lhs.static_cast();
-        let rhs: &DFPrimitiveArray<D> = rhs.static_cast();
-        let result_type = R::data_type();
-
-        let need_check = result_type.is_integer();
-
-        let result: DataColumn = try_binary_arithmetic_helper! {
-            lhs,
-            rhs,
+        let need_check = R::data_type().is_integer();
```
Comment (sundy-li) [inline common/functions/src/scalars/arithmetics/arithmetic_modulo.rs:59]: For `div`, `modulo`.

`D::default()` may cause div by zero error.

Comment (zhyass) [reply common/functions/src/scalars/arithmetics/arithmetic_modulo.rs:59]: > For `div`, `modulo`.
> 
> `D::default()` may cause div by zero error.

Yes, I use d::one()

File: common/datavalues/src/arrays/ops/pow.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+impl<T> Pow for DFPrimitiveArray<T>
+where
+    T: DFPrimitiveType,
+    DFPrimitiveArray<T>: ArrayCast,
+{
+    fn pow_f32(&self, exp: f32) -> DFFloat32Array {
+        let arr = self.cast_with_type(&DataType::Float32).expect("f32 array");
+        let arr = arr.f32().unwrap();
```
Comment (zhang2014) [inline common/datavalues/src/arrays/ops/pow.rs:37]: unwrap is unsafe?

Comment (zhyass) [reply common/datavalues/src/arrays/ops/pow.rs]: > unwrap is unsafe?

deleted

**Review by zhang2014 — APPROVED** [review]
Look good to me.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**